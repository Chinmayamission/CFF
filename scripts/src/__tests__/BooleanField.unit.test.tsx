import React from "react";
import { shallow, mount, render } from 'enzyme';
import CustomForm from "src/form/CustomForm";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders radio box', () => {
  let schema = {
    "type": "object",
    "title": "Form",
    "properties": {
      "donation": {
        "title": "Donation?",
        "description": "Our description of the donation",
        "type": "boolean"
      }
    }
  };
  let uiSchema = {
    "donation": {
      "ui:widget": "radio"
    }
  }
  const wrapper = render(
    <CustomForm schema={schema} uiSchema={uiSchema} />
  );
  // expect(wrapper.exists(".root_interests")).toEqual(true);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain("Donation?");
  expect(wrapper.text()).toContain("Our description of the donation");
}); 