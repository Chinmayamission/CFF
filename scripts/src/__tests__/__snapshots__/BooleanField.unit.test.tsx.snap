// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders radio box 1`] = `
<form
  class="rjsf"
>
  <div
    class="form-group field field-object"
  >
    <div
      class="container-fluid"
    >
      <h2
        class="ccmt-cff-form-title"
      >
        Form
      </h2>
      <div />
      <div
        class="row"
      >
        <div
          class="form-group field field-boolean  col-12"
        >
          <div>
            <label
              class="control-label"
              for="root_donation"
            >
              <span>
                Donation?
              </span>
            </label>
          </div>
          <div
            class="field-radio-group"
          >
            <div
              class="radio "
            >
              <label>
                <span>
                  <input
                    name="0.2706574174218739"
                    type="radio"
                    value="true"
                  />
                  <span>
                    yes
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radio "
            >
              <label>
                <span>
                  <input
                    name="0.2706574174218739"
                    type="radio"
                    value="false"
                  />
                  <span>
                    no
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div />
          <div />
        </div>
      </div>
    </div>
    <div />
    <div />
  </div>
  <p>
    <button
      class="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </p>
</form>
`;
